sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSink.scala

```diff
@@ -68,8 +68,15 @@ object FileStreamSink extends Logging {
     }
   }
 
-  def getMetadataLogPath(fs: FileSystem, path: Path, sqlConf: SQLConf): Path = {
-    val metadataDir = new Path(path, FileStreamSink.metadataDir)
+  def getMetadataLogPath(
+      hadoopConfiguration: Configuration,
+      path: Path,
+      sqlConf: SQLConf): Path = {
+    val metadataDir = sqlConf.streamingMetadataOutputPath match {
+      case Some(customPath) => new Path(customPath, FileStreamSink.metadataDir)
+      case None => new Path(path, FileStreamSink.metadataDir)
+    }
```

Comment:
This looks nice, but `streamingMetadataOutputPath` is `String` type, path is `Path` type, so I would need to do something like 

    val metadataPath = sqlConf.streamingMetadataOutputPath.getOrElse(path.toString)

that IMO is worse than match, because `path.toString` changes type and then is used back in `Path()`

WDYT?